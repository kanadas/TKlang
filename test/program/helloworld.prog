
tail :: [char] -> [char];
tail l = match l with {
    h:t -> t;
    [] -> []
};

stream main
input
state buffer = tail buffer
output 
    print = match buffer with {
        1@(h,t) -> h;
        _ -> '0'
        };
    exit = match buffer with {
        [_] -> true;
        2@() -> true;
        _ -> false
    }
initial 
    print = '0',
    buffer = "Hello World!",
    exit = false
