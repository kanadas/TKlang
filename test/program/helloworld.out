program/helloworld.prog
Hello World!